modyn/trainer_server/internal/dataset/key_sources/local_key_source.py

```diff
@@ -33,11 +33,11 @@ def get_num_data_partitions(self) -> int:
     def end_of_trigger_cleaning(self) -> None:
         self._trigger_sample_storage.clean_trigger_data(self._pipeline_id, self._trigger_id)
 
-    def __getstate__(self):
+    def __getstate__(self) -> Dict[str, Any]:
         state = self.__dict__.copy()
         del state["_trigger_sample_storage"]  # not pickable
         return state
 
-    def __setstate__(self, state):
+    def __setstate__(self, state: Dict[str, Any]) -> None:
```

Comment:
@MaxiBoether really curious how the CI passes on your PR :D
any repository founder privilege?